builtin/logical/aws/client.go

```diff
@@ -81,13 +79,19 @@ func (b *backend) getRootConfigs(ctx context.Context, s logical.Storage, clientT
 	credsConfig.HTTPClient = cleanhttp.DefaultClient()
 	credsConfig.Logger = logger
 
+	if config.Region != "" {
```

Comment:
This (probably) was the bug that was causing gov-cloud issues - we weren't pre-setting the region defined in the config, so it was getting lost.